src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,423 @@
+		while (oscap_string_iterator_has_more(product_ids)) {
+			const char *product_id = oscap_string_iterator_next(product_ids);
+			/*
+			if (cvrf_product_vulnerability_fixed(vuln, product_id)) {
+
+			}
+			else {
+
+				vulnerable = true;
+			}*/
+		}
+
+		oscap_string_iterator_free(product_ids);
+		vulnerable = false;
+	}
+	cvrf_vulnerability_iterator_free(it);
+
+	struct oscap_source *source = oscap_source_new_from_xmlDoc(doc, export_file);
+	int ret = oscap_source_save_as(source, NULL);
+	oscap_source_free(source);
+	cvrf_session_free(session);
+	return ret;
+}
+
+
+static const char *get_rpm_name_from_cvrf_product_id(struct cvrf_session *session, const char *product_id) {
+	const char *rpm_name = NULL;
+	struct cvrf_product_tree *tree = cvrf_model_get_product_tree(session->model);
+
+	struct oscap_iterator *branches = cvrf_product_tree_get_branches(tree);
+	while (oscap_iterator_has_more(branches)) {
+		struct cvrf_branch *branch = oscap_iterator_next(branches);
+		if (cvrf_branch_get_branch_type(branch) == CVRF_BRANCH_PRODUCT_VERSION) {
+			struct cvrf_product_name *full_name = cvrf_branch_get_cvrf_product_name(branch);
+
+			if (oscap_str_endswith(product_id, cvrf_product_name_get_product_id(full_name))) {
+				rpm_name = cvrf_product_name_get_cpe(full_name);
+				break;
+			}
+		}
+	}
+	oscap_iterator_free(branches);
+	return rpm_name;
+}
+
+static struct cvrf_rpm_attributes *parse_rpm_attributes_from_cvrf_product_id(struct cvrf_session *session,
+		const char *product_id) {
+
+	struct cvrf_rpm_attributes *attributes = cvrf_rpm_attributes_new();
+	attributes->full_package_name = strdup(get_rpm_name_from_cvrf_product_id(session, product_id));
+	char *package = strdup(strchr(product_id, ':')+1);
+	unsigned int index = 0;
+	while (index < strlen(package)) {
+		if (package[index] == ':')
+			break;
+		index++;
+	}
+	char *evr = &package[index-1];
+	package[index-2] = NULL;
+
+	attributes->evr_format = evr;
+	attributes->rpm_name = package;
+	return attributes;
+}
+
+
+bool cvrf_product_vulnerability_fixed(struct cvrf_vulnerability *vuln, char *product) {
+	struct cvrf_product_status_iterator *it = cvrf_vulnerability_get_product_statuses(vuln);
+	while (cvrf_product_status_iterator_has_more(it)) {
+		struct cvrf_product_status *stat = cvrf_product_status_iterator_next(it);
+		struct oscap_string_iterator *product_ids = cvrf_product_status_get_ids(stat);
+
+		while (oscap_string_iterator_has_more(product_ids)) {
+			const char *product_id = oscap_string_iterator_next(product_ids);
+
+			if (strcmp(product_id, product)) {
+				oscap_string_iterator_free(product_ids);
+				cvrf_product_status_iterator_free(it);
+				return true;
+			}
+		}
+		oscap_string_iterator_free(product_ids);
+	}
+	cvrf_product_status_iterator_free(it);
+
+	return false;
+}
+
+
+static char *get_oval_id_string(const char *type, int object_number) {
+	return oscap_sprintf("oval:org.open-scap.unix:%s:%d", type, object_number);
```

Comment:
This is just creating IDs for the objects, states, and tests that are generated during evaluation. The RPMInfo tests are included in the Unix family